### ISSUE TITLE
Conv1DTranspose?

### ISSUE THREAD

**Issue by hadaev8** [keras-team/keras#8705]
I want to make GAN for 1d object.
Whats best way to implement Conv1DTranspose?

[Here](https://stackoverflow.com/questions/44061208/how-to-implement-the-conv1dtranspose-in-keras) write 

> Do not always use transpose operation for it will consume a lot of time.

So probably use Conv2DTranspose is a bad idea.

[Here](https://github.com/fchollet/keras/issues/2822)

@fbranchaud-charron-miovision
> As Francois has explained multiple times already, a deconvolution layer is only a convolution layer with an upsampling. I don't think there is an official deconvolution layer. The result is the same.

But

@jwgu
> The deconvolution is not equivalent as "convolution + upsampling".

So, how best to do this? Conv2DTranspose? UpSampling1D? Something else?

**Comment by OtterOrder** [comment]
Hi,

I have the exact same problem.
I'm not sure why there is no Conv1DTranspose in the first place since it proved pretty convenient for image generation for instance.
Also, I tried using UpSampling1D but unfortunately, it takes more and more time the more I add layers until it finally crashes.

@hadaev8 Did you try with UpSampling1D? Did it work for you?

**Comment by hadaev8** [comment]
@OtterOrder 
not yet tried

**Comment by saber7165** [comment]
@OtterOrder and @hadaev8 , I have tried both UpSampling1D+Conv1D and also Conv2DTranspose with a 1D kernel on 1D data. They were pretty much the same in speed and accuracy.

**Comment by MatthewAwesome** [comment]
This feature has been implemented in tf.keras, (tf v2.2.0), [check it out!](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv1DTranspose)
